src/main/java/frc/robot/subsystems/Elevator/Elevator.java

```diff
@@ -0,0 +1,41 @@
+package frc.robot.subsystems.Elevator;
+
+import static frc.robot.GlobalConstants.ROBOT_MODE;
+
+import org.littletonrobotics.junction.Logger;
+import frc.robot.pioneersLib.subsystem.Subsystem;
+
+public class Elevator extends Subsystem<ElevatorStates> {
+    private static Elevator instance;
+
+    private ElevatorIO io;
+    private ElevatorIOInputsAutoLogged inputs;
+
+    private Elevator() {
+        super("Elevator", ElevatorStates.IDLE);
+        this.io = switch (ROBOT_MODE) {
+            case SIM -> new ElevatorIOSim();
+            case REAL -> new ElevatorIOReal();
+            case TESTING -> new ElevatorIOReal();
+            case REPLAY -> new ElevatorIOSim();
+        };
+        inputs = new ElevatorIOInputsAutoLogged();
+    }
+
+    public static Elevator getInstance() {
+        if (instance == null) {
+            instance = new Elevator();
+        }
+        return instance;
+    }
+
+    @Override
+    protected void runState() {
+        io.setHeightGoalpoint(getState().getTargetHeight());
+        io.runElevator();
```

Comment:
where is your zeroing function? elevator should run motors downwards at a constant speed. Then the elevator hits a limit switch (magnetic) and you zero the encoders. U can do the logic however u want. For sim just make it zero on init after like 1 second. For creating a limit switch, u just create a DIO object (basically how those ir sensors/beam breaks work) and just check if its true